compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -681,11 +681,20 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
             Some(self_ty),
```

Comment:
^we were probably also allowing `?std::ops::Fn<T>` but i dont care to write a test for that.